ui/page/wallet_page.go

```diff
@@ -129,9 +132,16 @@ func (pg *WalletPage) ID() string {
 }
 
 func (pg *WalletPage) OnResume() {
+	pg.loadWalletAndAccounts()
+}
+
+func (pg *WalletPage) loadWalletAndAccounts() {
 	wallets := pg.WL.SortedWalletList()
 
+	pg.listLock.Lock()
 	pg.listItems = make([]*walletListItem, 0)
```

Comment:
Instead of doing this, store the `walletListItem` in a temp local slice and copy to locked slice when done.
```suggestion
	listItems := make([]*walletListItem, 0)
```